### PR TITLE
Browser: Implement DOM Inspector for multi-process mode

### LINKED ISSUE #8935: Browser: Inspector does not work in multi-process mode
Using single-process mode when you need to inspect things is workable, but ideally we would be able to inspect in multi-process mode too.

### PR DESCRIPTION
**Pull request by AtkinsSJ**
...which is now the *only* mode!

This fixes #8935.

The tl;dr of this is: Each DOM Node now has a unique u64 ID, which the DOM Inspector can use to communicate with the OOPWV about that Node. This allowed the "Inspect DOM Tree" and "Inspect Element" actions, and displaying of style properties, to be reimplemented as they worked before in single-process mode. :^)

There is still quite a bit of work to do on the Inspector, (listed below,) but I wanted to get this out first so people can use it again.

- Selecting the `<body>` node by default, if it exists
- Reloading the DOM JSON when the DOM changes. (There are some crashes with calling `document.body.innerHTML = ""` with the inspector open, haven't investigated the cause though.)
- Reloading the style data when it changes
- Inspecting inside `<iframe>`, `<template>`, and similar nodes
- Getting the Computed Values to display according to the spec
- Make the style values' `to_string()` spec-compliant, and making them editable
- Figure out what should go in the Layout tab and reimplement that

### CODE REVIEW

File: Userland/Libraries/LibWeb/DOM/Node.cpp [hunk 1]
```diff
@@ -30,10 +31,56 @@
...
 namespace Web::DOM {
 
+class NodeDirectory {
+public:
+    NodeDirectory()
+        : m_nodes_by_id()
+    {
+    }
+
+    u64 create_node_id(Node* node)
```
Comment (awesomekling) [inline Userland/Libraries/LibWeb/DOM/Node.cpp:41]: Seems like a decent usecase for `AK::IDAllocator` :^)

File: Userland/Libraries/LibWeb/DOMTreeModel.cpp [hunk 2]
```diff
@@ -166,4 +167,22 @@ void DOMTreeModel::map_dom_nodes_to_parent(JsonObject const* parent, JsonObject
     });
 }
 
+GUI::ModelIndex DOMTreeModel::index_for_node(u64 node_id)
+{
+    auto maybe_node = m_node_id_to_dom_node_map.get(node_id);
+    if (maybe_node.has_value()) {
+        auto* node = maybe_node.value();
```
Comment (awesomekling) [inline Userland/Libraries/LibWeb/DOMTreeModel.cpp:174]: I usually write these as
```cpp
auto* node = m_node_id_to_dom_node_map.get(node_id).value_or(nullptr);
```

File: Userland/Libraries/LibWeb/DOMTreeModel.h [hunk 3]
```diff
@@ -33,6 +33,8 @@ class DOMTreeModel final : public GUI::Model {
     virtual GUI::ModelIndex index(int row, int column, const GUI::ModelIndex& parent = GUI::ModelIndex()) const override;
     virtual GUI::ModelIndex parent_index(const GUI::ModelIndex&) const override;
 
+    GUI::ModelIndex index_for_node(u64 node_id);
```
Comment (awesomekling) [inline Userland/Libraries/LibWeb/DOMTreeModel.h:36]: Can this be `const` qualified?

File: Userland/Libraries/LibWeb/DOM/Node.cpp (hunk 1 above)
Comment (AtkinsSJ) [reply Userland/Libraries/LibWeb/DOM/Node.cpp:41]: Oooh, that's very handy! Thanks!

File: Userland/Libraries/LibWeb/DOMTreeModel.h (hunk 3 above)
Comment (AtkinsSJ) [reply Userland/Libraries/LibWeb/DOMTreeModel.h:36]: Yup, it can. My brain still doesn't think in `const`-ness yet.
